### PR TITLE
refactore(leverage): rename ExchangeToken

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

ref: https://github.com/umee-network/umee/pull/2167#discussion_r1275259898

+ fixed go mocks in metoken (they were outdated)
+ fixed intest in quota (it was using wrong constructor)

### CODE REVIEW

**Review by toteki — CHANGES_REQUESTED** [review]
Let's think of a better name

**Comment by toteki** [comment]
Maybe `ToToken()` and `ToUToken()`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
   - `/umee/ugov/v1/min-gas-price` --> `/umee/ugov/v1/min_gas_price`
   - `/umee/ugov/v1/emergency-group` --> `/umee/ugov/v1/emergency_group`
   - `/umee/uibc/v1/all-outflows` --> `/umee/uibc/v1/all_outflows`
+- [2180](https://github.com/umee-network/umee/pull/2180) Rename leverage `Kepeer.ExchangeToken -> ToUToken`, `Kepeer.ExchangeUToken -> ToTokenRate` and `Kepeer.ExchangeTokens -> ToUTokens`.
```
Comment (toteki) [inline CHANGELOG.md:77]:
```suggestion
- [2180](https://github.com/umee-network/umee/pull/2180) Rename leverage `Keeper.ExchangeToken -> ToUToken`, `Keeper.ExchangeUToken -> ToToken` and `Keeper.ExchangeTokens -> ToUTokens`.
```

File: x/leverage/keeper/collateral.go [hunk 2]
```diff
@@ -251,7 +251,7 @@ func (k Keeper) ModuleMaxWithdraw(ctx sdk.Context, spendableUTokens sdk.Coin) (s
 
 	// Get module collateral for the uDenom
 	totalCollateral := k.GetTotalCollateral(ctx, spendableUTokens.Denom)
-	totalTokenCollateral, err := k.ExchangeUTokens(ctx, sdk.NewCoins(totalCollateral))
+	totalTokenCollateral, err := k.Tokens2uTokensRate(ctx, sdk.NewCoins(totalCollateral))
```
Comment (toteki) [inline x/leverage/keeper/collateral.go:254]: Still needs a rename

File: x/leverage/keeper/exchange_rate.go [hunk 3]
```diff
@@ -6,9 +6,9 @@ import (
 	"github.com/umee-network/umee/v5/x/leverage/types"
 )
 
-// ExchangeToken converts an sdk.Coin containing a base asset to its value as a
-// uToken.
-func (k Keeper) ExchangeToken(ctx sdk.Context, token sdk.Coin) (sdk.Coin, error) {
+// ToUToken returns a uToken amount user would receive when supplying the token.
+// Returns error if the token is not a Token.
```
Comment (toteki) [inline x/leverage/keeper/exchange_rate.go:10]:
```suggestion
// ToUToken converts a token to uTokens at the current uToken exchange rate.
// This is the uToken amount a user would receive when supplying the token.
// Returns error if the input is not a Token.
```

File: x/leverage/keeper/exchange_rate.go [hunk 4]
```diff
@@ -19,14 +19,13 @@ func (k Keeper) ExchangeToken(ctx sdk.Context, token sdk.Coin) (sdk.Coin, error)
...
-
 	uTokenAmount := toDec(token.Amount).Quo(exchangeRate).TruncateInt()
 	return sdk.NewCoin(uTokenDenom, uTokenAmount), nil
 }
 
-// ExchangeUToken converts an sdk.Coin containing a uToken to its value in a base
-// token.
-func (k Keeper) ExchangeUToken(ctx sdk.Context, uToken sdk.Coin) (sdk.Coin, error) {
+// ToToken returns a token amount user would receive when supplying the uToken.
+// Returns error if the uToken is not a uToken.
```
Comment (toteki) [inline x/leverage/keeper/exchange_rate.go:27]:
```suggestion
// ToToken converts a uToken to tokens at the current uToken exchange rate.
// This is the token amount a user would receive when withdrawing the uToken.
// Returns error if the input is not a uToken.
```

File: x/leverage/keeper/exchange_rate.go [hunk 5]
```diff
@@ -42,16 +41,16 @@ func (k Keeper) ExchangeUToken(ctx sdk.Context, uToken sdk.Coin) (sdk.Coin, erro
 	return sdk.NewCoin(tokenDenom, tokenAmount), nil
 }
 
-// ExchangeUTokens converts an sdk.Coins containing uTokens to their values in base
+// Tokens2uTokensRate converts an sdk.Coins containing uTokens to their values in base
```
Comment (toteki) [inline x/leverage/keeper/exchange_rate.go:44]:
```suggestion
// ToToken converts a uTokens to tokens at their current uToken exchange rates.
// These are the token amounts a user would receive when withdrawing the uTokens.
// Returns error if any of the inputs are uTokens.
```

**Review by toteki — COMMENTED** [review]
a little cleanup

File: x/leverage/keeper/exchange_rate.go [hunk 6]
```diff
@@ -42,16 +43,18 @@ func (k Keeper) ExchangeUToken(ctx sdk.Context, uToken sdk.Coin) (sdk.Coin, erro
 	return sdk.NewCoin(tokenDenom, tokenAmount), nil
 }
 
-// ExchangeUTokens converts an sdk.Coins containing uTokens to their values in base
+// ToToken converts an sdk.Coins containing uTokens to tokens using uToken exchange rates.
+// These are the token amounts a user would receive when withdrawing the uTokens.
+// Returns error if any of the inputs are uTokens.
 // tokens.
-func (k Keeper) ExchangeUTokens(ctx sdk.Context, uTokens sdk.Coins) (sdk.Coins, error) {
+func (k Keeper) Tokens2uTokensRate(ctx sdk.Context, uTokens sdk.Coins) (sdk.Coins, error) {
```
Comment (toteki) [inline x/leverage/keeper/exchange_rate.go:50]:
```suggestion
func (k Keeper) ToUTokens(ctx sdk.Context, uTokens sdk.Coins) (sdk.Coins, error) {
```

**Review by kosegor — APPROVED** [review]
approving `x/metoken` module part

File: x/leverage/keeper/exchange_rate.go [hunk 7]
```diff
@@ -42,16 +41,17 @@ func (k Keeper) ExchangeUToken(ctx sdk.Context, uToken sdk.Coin) (sdk.Coin, erro
 	return sdk.NewCoin(tokenDenom, tokenAmount), nil
 }
 
-// ExchangeUTokens converts an sdk.Coins containing uTokens to their values in base
-// tokens.
-func (k Keeper) ExchangeUTokens(ctx sdk.Context, uTokens sdk.Coins) (sdk.Coins, error) {
+// ToToken returns list of Tokens in the amount a user would receive when withdrawing the
+// list of uTokens.
+// Returns error if any of the inputs are uTokens.
```
Comment (toteki) [inline x/leverage/keeper/exchange_rate.go:46]:
```suggestion
// ToTokens returns list of Tokens in the amount a user would receive when withdrawing the
// list of uTokens.
// Returns error if any of the inputs are tokens.
```

**Review by toteki — APPROVED** [review]

**Pull request merged**